lib/modules/manager/composer/schema.ts

```diff
@@ -26,26 +26,34 @@ export type ComposerRepo = z.infer<typeof ComposerRepo>;
 export const GitRepo = z.object({
   type: z.enum(['vcs', 'git']).transform(() => 'git' as const),
   url: z.string(),
+  name: z.string().optional(),
 });
 export type GitRepo = z.infer<typeof GitRepo>;
 
 export const PathRepo = z.object({
   type: z.literal('path'),
   url: z.string(),
+  name: z.string().optional(),
 });
 export type PathRepo = z.infer<typeof PathRepo>;
 
+export const PackageRepo = z.object({
+  type: z.literal('package'),
+});
```

Comment:
We don't actually support it, but seems like we can do it for GitHub-based deps, like we do for Bazel